### PR TITLE
Extract action partials from `admin/accounts` show buttons/actions area

### PR DESCRIPTION
**Pull request by mjankowski**
As noted on https://github.com/mastodon/mastodon/pull/32154

> [...] this partial in particular is super super dense and hard to reason through all the scenarios -- could benefit from either/both of a) getting rid of all the trailing conditional guard clause can, b) in some spots, even more partials for the high level logic.

Previously here, the main show view called one buttons partial with the account and (sometimes present) deletion request, and that partial then handled two main paths -- stuff for suspended users, stuff for everyone else. Within the "everyone else" path, its just a bunch of dense nested logic around what state the account is in, what actions are relevant, and if the account looking at it can actually do the relevant actions.

I find that partially basically impossible to parse due to the sheer length, but also the nested conditions and trailing guard clauses and so on. Approach here is a mix of things...

- In main show view, pull the "suspended or not" distinction back up into the top level view, and only pass the deletion request when needed into a different partial. Break down the other stuff into a "core" and "extras" distinction as well
- Extract each logical area or feature into their own partial ... generally speaking the "is this relevant" check stays in the calling location and the "can this person do this" check goes in the partial.

While this is still pretty dense in parts, I think it's a net improvement. Possible future further improvements:

- The main account show view here is still pretty sprawling ... but has some obvious possible partial extractions -- 1) the whole fields/notes section on top, 2) the warnings/strikes area near bottom, 3) the notes listing and form near bottom -- maybe more
- Within the suspended path, I cleaned up the muted hint path a little bit, but those two long ternary lines could probably be improved further, moved to helpers, etc
- Since these are only ever used from the one view, we could put a prepend_view_paths call in the controller to make all the partial render lines here shorter (omit path)
- If we ever revisit the i18n lazy lookup policy, these would all become more readable by having shorter i18n keys

Curious to hear thoughts. I tried to keep this to smallish commits for easier review, but it still may be challenging. The goal was to preserve logic/markup, so definitely raise any issues if you think I broke something. I guess in theory I could try to extract things from this and step it through one smaller PR at a time - but I also wanted to get the whole idea up in advance.

### CODE REVIEW

**Comment by oneiros** [comment]
Oh my. I can totally understand the motivation here. I worked on several projects that ended up with a partial like that. I agree it is not pretty and a real pain to make sense of.

But I am not convinced these changes improve the situation.

To me, the problem is not that the buttons partial is too long per se but rather the nested conditional logic that is hard to parse. I think one could argue these changes make that worse, because they do not change that logic at all. They only spread it out over several files. This can make it even harder to parse (e.g. when looking at `app/views/admin/accounts/actions/_confirmation.html.haml` you now lack the context that the account is local and not confirmed), but to me it also makes it harder to find (so many files to look through). Oh, and it makes it harder to change the order of buttons.

Sorry, I do not have a better solution, but I wonder if something could be done with those conditionals. For example, instead of having:

```ruby
if account.local?
  if account.user_pending?
    # ...
  if !account.user_confirmed?
    # ...
```

we could maybe say

```ruby
if account.local? && account.user_pending?
   # ...
if account.local? && !account.user_confirmed?
```

or even hide this behind new methods on either the model, a presenter or as helper methods:

```ruby
if account_local_and_pending?
  # ...
if account_local_and_unconfirmed?
  # ...
```

(Yes, that means an extra call to `account.local?` but that should be cheaper than rendering extra partials.)

I agree with moving the "suspended or not" distinction back up into the top level view.

If as a last measure we consistently use the postfix `if` to check permissions, which is currently done on most but not all buttons, then we might end up with a smaller and more uniform buttons partial.

But that is just a vague idea, not sure if it would work in practice.

**Comment by mjankowski** [comment]
Thanks for all that ... I agree with the spirit of the feedback. In my opinion this change is SLIGHTLY better, but I agree it's not some incredible improvement. The nature of what's happening here is hard to make nice  and tidy.

Why don't I do a separate PR to do nothing but the extraction of the suspended path back into the view, and then maybe a pass at some helper methods. Will leave this open and keep rebased and we can see how that evolves.
